File: QadeerApp.Web/Modules/ServerTypes/Administration/TrainingCourseForm.ts
import { BooleanEditor, initFormType, LookupEditor, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface TrainingCourseForm {
    DepartmentId: LookupEditor;
    SpecializationId: LookupEditor;
    Name: StringEditor;
    Code: StringEditor;
    IsActive: BooleanEditor;
}

export class TrainingCourseForm extends PrefixedContext {
    static readonly formKey = 'Administration.TrainingCourse';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!TrainingCourseForm.init) {
            TrainingCourseForm.init = true;

            var w0 = LookupEditor;
            var w1 = StringEditor;
            var w2 = BooleanEditor;

            initFormType(TrainingCourseForm, [
                'DepartmentId', w0,
                'SpecializationId', w0,
                'Name', w1,
                'Code', w1,
                'IsActive', w2
            ]);
        }
    }
}

File: QadeerApp.Web/Modules/Administration/TrainingCourse/TrainingCourseForm.cs
namespace QadeerApp.Administration.Forms;

[FormScript("Administration.TrainingCourse")]
[BasedOnRow(typeof(TrainingCourseRow), CheckNames = true)]
public class TrainingCourseForm
{
    public int DepartmentId { get; set; }
    public int? SpecializationId { get; set; }
    public string Name { get; set; }
    public string Code { get; set; }
    public bool IsActive { get; set; }
}
